### PR TITLE
Add XSS support for Freemarker post 2.3.24-incubating

### PR DESCRIPTION
**Pull request by Mariovido**
# What Does This Do
Adds support to the detection of XSS in the Freemarker library from the 2.3.24-incubating version

# Motivation
Being able to detect XSS in the library of Freemarker

# Additional Notes
The PR that adds support to oldest versions is this one --> https://github.com/DataDog/dd-trace-java/pull/7497

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [APPSEC-11285]

[APPSEC-11285]: https://datadoghq.atlassian.net/browse/APPSEC-11285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/instrumentation/freemarker-2.3.24/src/main/java/datadog/trace/instrumentation/freemarker24/DollarVariableDatadogAdvice.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import datadog.trace.api.iast.sink.XssModule;
+import freemarker.core.DollarVariable24Helper;
+import freemarker.core.Environment;
+import net.bytebuddy.asm.Advice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class DollarVariableDatadogAdvice {
+
+  protected static final Logger log = LoggerFactory.getLogger(DollarVariableDatadogAdvice.class);
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/instrumentation/freemarker-2.3.24/src/main/java/datadog/trace/instrumentation/freemarker24/DollarVariableDatadogAdvice.java:15]: I think we can remove this one.

File: dd-smoke-tests/springboot-freemarker/src/main/resources/templates/freemarker-2.3.9.ftlh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<html>
```
Comment (manuel-alvarez-alvarez) [inline dd-smoke-tests/springboot-freemarker/src/main/resources/templates/freemarker-2.3.9.ftlh:1]: Can you add a test case using the [<#escape](https://freemarker.apache.org/docs/ref_directive_escape.html) directive, to validate that we don't trigger the vuln?

File: dd-smoke-tests/springboot-freemarker/src/main/resources/templates/freemarker-2.3.24.ftlh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<#ftl output_format="HTML" auto_esc=false>
```
Comment (manuel-alvarez-alvarez) [inline dd-smoke-tests/springboot-freemarker/src/main/resources/templates/freemarker-2.3.24.ftlh:1]: Can you add another test case with `auto_esc=true` to ensure that we don't trigger the vuln?

**Review by manuel-alvarez-alvarez — APPROVED** [review]
LGTM, good job!

File: dd-java-agent/instrumentation/freemarker-2.3.24/src/main/java/datadog/trace/instrumentation/freemarker24/DollarVariableDatadogAdvice.java (hunk 1 above)
Comment (Mariovido) [reply dd-java-agent/instrumentation/freemarker-2.3.24/src/main/java/datadog/trace/instrumentation/freemarker24/DollarVariableDatadogAdvice.java]: Removed!

**Pull request merged**